theano/gpuarray/opt.py

```diff
@@ -845,10 +848,14 @@ def local_gpu_pdbbreakpoint_op(node):
 def local_gpua_lazy_ifelse(op, context_name, inputs):
     if op.gpu:
         return
+    if isinstance(inputs[0].type, GpuArrayType):
+        return
```

Comment:
Exactly. line 849, 850 says that. But, when i made this change, the cycles in graph doesn't exist anymore. This caused cycle:(line 309 inside gpuarray/tests/test_opt.py

```
    t = tensor.fmatrix()
    f = theano.function([t], t[3:4], mode=mode_with_gpu)